### DISCUSSION TITLE
PAGPlayer+GLSurfaceView的使用方式，出现native内存不释放的问题

### DISCUSSION THREAD

**Discussion by cerero1** [Tencent/libpag#2597] [Q&A] [answered]
你好大佬们，最近小弟使用PAGPlayer+GLSurfaceView的方式进行demo演示，发现native内存有不释放的情况，具体如下：
![2a1600fda2ed4afa82107ee9844cd313](https://github.com/user-attachments/assets/de27e4b0-7e39-4e1e-9fd5-42a4523e1a2d)
native涨了近20MB不释放，假如强制触发GC，native依然保持10MB+的增量

我是这样使用PAGPlayer的：
![68993ccfaf58cdecb0e99d936a2b2baa](https://github.com/user-attachments/assets/40214b48-16fc-410b-ae71-87ff8d2659b8)
![6fef7b139efcbc8dd80ac7552ee152b2](https://github.com/user-attachments/assets/14bc19e8-0dfa-4aef-b05e-f5155ac318d9)
![e91e0f9ab17f6039f1eca800bd865b96](https://github.com/user-attachments/assets/bbb5069d-6df8-4828-8a36-a7bae8fb2480)

![aa67befaf2c3eb195c5c2c13a6704e24](https://github.com/user-attachments/assets/aeb04c85-1552-4e72-972b-de95b94c33ef)
![f3c9e6e979419c52b453a8f929fe78f1](https://github.com/user-attachments/assets/fd8e1c20-696e-4bf3-a07e-4f56e85699aa)

**Comment by kevingpqi123** [comment, marked as answer]
释放的时候增加上 PAGSurface 的 freeCache，可以理解释放 pag 申请的一些内存，这里和 pag 没啥关系， GLSurfaceView 的 OpenGL 环境是系统层面提供，相关释放机制由系统层面处理
